### ISSUE TITLE
Open alsoft fails to connect to output port.

### ISSUE THREAD

**Issue by MythicalNewt** [kcat/openal-soft#907]
`[ALSOFT] (EE) Failed to connect output port "alsoft:channel_1" to "SteelSeries Arctis 5 Game:playback_FL"
[ALSOFT] (EE) Failed to connect output port "alsoft:channel_2" to "SteelSeries Arctis 5 Game:playback_FR"
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)`

openal version: 1.23.1-1

**Comment by kcat** [comment]
Can you provide a full trace log? Set the `ALSOFT_LOGLEVEL` environment variable to `3` when you run the app to get a more detailed log output. Are you purposely using the JACK backend, or is it being autoselected? If you're purposely using JACK, are you able to connect the ports manually using a patchbay or graph UI app? If you have other devices, do they work?

**Comment by MythicalNewt** [comment]
`[ALSOFT] (II) Initializing library v1.21.1-2290817 master
[ALSOFT] (II) Supported backends: jack, pulse, alsa, oss, port, null, wave
[ALSOFT] (II) Loading config /etc/openal/alsoft.conf...
[ALSOFT] (II) Loading config /etc/xdg/alsoft.conf...
[ALSOFT] (II) Loading config /home/shreyas/.alsoftrc...
[ALSOFT] (II) Loading config /home/shreyas/.config/alsoft.conf...
[ALSOFT] (II) Got binary: /usr/lib/jvm/java-17-openjdk/bin, java
[ALSOFT] (II) Loading config /usr/lib/jvm/java-17-openjdk/bin/alsoft.conf...
[ALSOFT] (II) Key disable-cpu-exts not found
[ALSOFT] (II) Vendor ID: "AuthenticAMD"
[ALSOFT] (II) Name: "AMD Ryzen 7 2700X Eight-Core Processor"
[ALSOFT] (II) Extensions: +SSE +SSE2 +SSE3 +SSE4.1
[ALSOFT] (II) Key rt-prio not found
[ALSOFT] (II) Key resampler not found
[ALSOFT] (II) Key trap-al-error not found
[ALSOFT] (II) Key trap-alc-error not found
[ALSOFT] (II) Key reverb/boost not found
[ALSOFT] (II) Key drivers not found
[ALSOFT] (II) Key jack/spawn-server not found
[ALSOFT] (II) Initialized backend "jack"
[ALSOFT] (II) Added "jack" for playback
[ALSOFT] (II) Key pulse/spawn-server not found
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (II) Initialized backend "pulse"
[ALSOFT] (II) Added "pulse" for capture
[ALSOFT] (II) Key excludefx not found
[ALSOFT] (II) Key default-reverb not found
[ALSOFT] (II) Key jack/custom-devices not found
[19:00:13] [Render thread/INFO]: Hardware information:
[19:00:13] [Render thread/INFO]: CPU: 16x AMD Ryzen 7 2700X Eight-Core Processor
[19:00:13] [Render thread/INFO]: GPU: AMD Radeon RX 590 Series (polaris10, LLVM 16.0.6, DRM 3.52, 6.4.12-zen1-1-zen) (Supports OpenGL 4.6 (Core Profile) Mesa 23.1.6-arch1.4)
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key channels not found
[ALSOFT] (II) Key sample-type not found
[ALSOFT] (II) Key frequency not found
[ALSOFT] (II) Key period_size not found
[ALSOFT] (II) Key periods not found
[ALSOFT] (II) Key sources not found
[ALSOFT] (II) Key slots not found
[ALSOFT] (II) Key sends not found
[ALSOFT] (II) Key ambi-format not found
[ALSOFT] (II) Created device 0x7f56f4ecec00, "JACK Default"
[ALSOFT] (II) Key hrtf-paths not found
[ALSOFT] (II) Searching /home/shreyas/.local/share/PrismLauncher/instances/BatuSMP/.minecraft for *.mhr
[ALSOFT] (II) Searching /home/shreyas/.local/share/openal/hrtf for *.mhr
[ALSOFT] (II) Searching /home/shreyas/.local/share/flatpak/exports/share/openal/hrtf for *.mhr
[ALSOFT] (II) Searching /var/lib/flatpak/exports/share/openal/hrtf for *.mhr
[ALSOFT] (II) Searching /usr/local/share/openal/hrtf for *.mhr
[ALSOFT] (II) Searching /usr/share/openal/hrtf for *.mhr
[ALSOFT] (II)  got /usr/share/openal/hrtf/Default HRTF.mhr
[ALSOFT] (II) Adding file entry "/usr/share/openal/hrtf/Default HRTF.mhr"
[ALSOFT] (II) Adding built-in entry "!1_Built-In HRTF"
[ALSOFT] (II) Key default-hrtf not found
[ALSOFT] (II) ALC_HRTF_SOFT = 1
[ALSOFT] (II) ALC_HRTF_ID_SOFT = 0
[ALSOFT] (II) Key frequency not found
[ALSOFT] (II) Key period_size not found
[ALSOFT] (II) Key periods not found
[ALSOFT] (II) Key sources not found
[ALSOFT] (II) Key sends not found
[ALSOFT] (II) Key hrtf not found
[ALSOFT] (II) Pre-reset: *Stereo, Float32, 44100hz, 882 / 2646 buffer
[ALSOFT] (II) Key jack/buffer-size not found
[ALSOFT] (II) Post-reset: Stereo, Float32, 48000hz, 1024 / 2048 buffer
[ALSOFT] (II) Key stereo-mode not found
[ALSOFT] (II) Loading /usr/share/openal/hrtf/Default HRTF.mhr...
[ALSOFT] (II) Detected data set format v3
[ALSOFT] (II) Loaded HRTF Default HRTF for sample rate 48000hz, 64-sample filter
[ALSOFT] (II) Key hrtf-size not found
[ALSOFT] (II) Key hrtf-mode not found
[ALSOFT] (II) 1st order + Full HRTF rendering enabled, using "Default HRTF"
[ALSOFT] (II) Channel config, Main: 4, Real: 2
[ALSOFT] (II) Allocating 6 channels, 24576 bytes
[ALSOFT] (II) Skipped delay: 7.75, new max delay: 24.75, FIR length: 89
[ALSOFT] (II) Key decoder/nfc not found
[ALSOFT] (II) Max sources: 256 (255 + 1), effect slots: 64, sends: 2
[ALSOFT] (II) Key dither not found
[ALSOFT] (II) Key dither-depth not found
[ALSOFT] (II) Dithering disabled
[ALSOFT] (II) Key output-limiter not found
[ALSOFT] (II) Output limiter disabled
[ALSOFT] (II) Fixed device latency: 5333333ns
[ALSOFT] (II) Key jack/connect-ports not found
[ALSOFT] (EE) Failed to connect output port "alsoft:channel_1" to "SteelSeries Arctis 5 Game:playback_FL"
[ALSOFT] (EE) Failed to connect output port "alsoft:channel_2" to "SteelSeries Arctis 5 Game:playback_FR"
[ALSOFT] (II) Key jack/buffer-size not found
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (II) ALC_MAX_AUXILIARY_SENDS = 4
[ALSOFT] (II) Key frequency not found
[ALSOFT] (II) Key period_size not found
[ALSOFT] (II) Key periods not found
[ALSOFT] (II) Key sources not found
[ALSOFT] (II) Key sends not found
[ALSOFT] (II) Key hrtf not found
[ALSOFT] (II) Pre-reset: *Stereo, Float32, 44100hz, 882 / 2646 buffer
[ALSOFT] (II) Key jack/buffer-size not found
[ALSOFT] (II) Post-reset: Stereo, Float32, 48000hz, 1024 / 2048 buffer
[ALSOFT] (II) Key stereo-mode not found
[ALSOFT] (II) HrtfStore 0x7f56f502e6a0 decreasing refcount to 0
[ALSOFT] (II) Unloading unused HRTF /usr/share/openal/hrtf/Default HRTF.mhr
[ALSOFT] (II) Key cf_level not found
[ALSOFT] (II) Key stereo-encoding not found
[ALSOFT] (II) Stereo rendering
[ALSOFT] (II) Channel config, Main: 3, Real: 2
[ALSOFT] (II) Allocating 5 channels, 20480 bytes
[ALSOFT] (II) Enabling single-band first-order ambisonic decoder
[ALSOFT] (II) Max sources: 256 (255 + 1), effect slots: 64, sends: 4
[ALSOFT] (II) Key dither not found
[ALSOFT] (II) Key dither-depth not found
[ALSOFT] (II) Dithering disabled
[ALSOFT] (II) Key output-limiter not found
[ALSOFT] (II) Output limiter disabled
[ALSOFT] (II) Fixed device latency: 0ns
[19:00:15] [Render thread/INFO]: OpenAL initialized on device JACK Default
[19:00:15] [Render thread/INFO]: Initializing Sound Physics
[19:00:15] [Render thread/INFO]: EFX Extension recognized
[19:00:15] [Render thread/INFO]: Max auxiliary sends: 4
[19:00:15] [Render thread/INFO]: Aux slot 1 created
[19:00:15] [Render thread/INFO]: Aux slot 2 created
[19:00:15] [Render thread/INFO]: Aux slot 3 created
[19:00:15] [Render thread/INFO]: Aux slot 4 created
[19:00:15] [Render thread/INFO]: EFX ready
[19:00:15] [Render thread/INFO]: Sound engine started
[ALSOFT] (II) Key jack/connect-ports not found
[ALSOFT] (EE) Failed to connect output port "alsoft:channel_1" to "SteelSeries Arctis 5 Game:playback_FL"
[ALSOFT] (EE) Failed to connect output port "alsoft:channel_2" to "SteelSeries Arctis 5 Game:playback_FR"
[ALSOFT] (II) Key jack/buffer-size not found
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (II) Increasing allocated voices to 256
[ALSOFT] (II) Key volume-adjust not found
[ALSOFT] (II) Created context 0x7f56f4a619a0
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
[ALSOFT] (II) Key jack/custom-devices not found
`

I tried cleaning up the minecraft log as much as possible

**Comment by MythicalNewt** [comment]
https://pastebin.com/ffUcpUYL

I tried cleaning up the minecraft console as much as possible and this is the sorted log

If you want to see the full log its here: https://pastebin.com/60Y6gnGa

**Comment by kcat** [comment]
1.21.1 is an older version, which unfortunately has poor interaction with PipeWire (PipeWire has JACK compatibility, which OpenAL Soft would previously only consider advanced users to have JACK). Though I have no idea at all why `jack_connect` would fail to connect ports, especially if you can do it manually, and I wouldn't know where to start looking.

One thing you can do to get it working is set up a config file to either disable the JACK backend, or prioritize PipeWire/PulseAudio. In `~/.config/alsoft.conf`, add
```
[general]
drivers=-jack, # comma is important
```
to disable the JACK backend and use the next available (PulseAudio for 1.21.1, PipeWire for later versions). Or alternatively
```
drivers=pipewire,pulse,
```
to prioritize PipeWire (on versions it's available), followed by PulseAudio, then any other remaining backend.

**Comment by MythicalNewt** [comment]
Turns out jack was the problem,
I was going through documentation and saw drivers and disabled jack sometime ago and came to report my findings.

Thank you for getting back so quickly
